controllers/volumereplicationgroup_controller.go

```diff
@@ -1718,22 +1744,24 @@ func getStatusStateFromSpecState(state ramendrv1alpha1.ReplicationState) ramendr
 //    VRG.conditions.Available.Status = false
 //    VRG.conditions.Available.Reason = Progressing
 //
-func (v *VRGInstance) updateVRGConditions() {
+func (v *VRGInstance) updateVRGDataReadyCondition() {
 	vrgReady := true
 	vrgProgressing := false
 
 	for _, protectedPVC := range v.instance.Status.ProtectedPVCs {
 		dataReadyCondition := findCondition(protectedPVC.Conditions, VRGConditionTypeDataReady)
 		if dataReadyCondition == nil {
 			vrgReady = false
-
+			// When will we hit this condition? If it is due to a race condition,
+			// why treat it as an error instead of progressing?
```

Comment:
If race condition is the only possible situation where this can happen, then yes I agree that reporting it as Progressing instead of error might be more appropriate. Since we are not certain about what exactly could be the reason for not able to find the correct condition, it is chosen to be error.

If say those conditions were put into a queue and someone processes the items from queue before properly publishing the conditions, then we would know that it is still being processed and we are looking for it before it is processed and hence mark it as progressing. Not being able to get the exact cause for the inability to find the condition is why error is chosen.